prediction_market_agent/agents/microchain_agent/market_functions.py

```diff
@@ -105,23 +100,63 @@ def __init__(
     def example_args(self) -> list[str]:
         return [get_example_market_id(self.market_type)]
 
+
+class PredictProbabilityForQuestion(PredictProbabilityForQuestionBase):
+    """
+    Uses the prediction_prophet library to make a prediction.
+    """
+
+    def __init__(
+        self,
+        market_type: MarketType,
+        keys: APIKeys,
+        model: str = "gpt-3.5-turbo",
+    ) -> None:
+        self.model = model
+        super().__init__(market_type=market_type, keys=keys)
+
+    @property
+    def description(self) -> str:
+        return self._description
+
     def __call__(self, market_id: str) -> str:
         question = self.market_type.market_class.get_binary_market(
             id=market_id
         ).question
-        response: MechResponse = self.mech_request(question, self.mech_tool)
-        return str(response.p_yes)
+        research = prophet_research(
+            goal=question,
+            use_summaries=False,
+            model=self.model,
+            openai_api_key=self.keys.openai_api_key,
+            tavily_api_key=self.keys.tavily_api_key,
+        )
+        prediction = prophet_make_prediction(
+            market_question=question,
+            additional_information=research,
+            engine=self.model,
+            temperature=0,
+            api_key=self.keys.openai_api_key,
+        )
+        if prediction.outcome_prediction is None:
+            return "Failed to make a prediction."
```

Comment:
I wonder if it's best to throw an error? Otherwise the agent might take this and interpret this somehow as `p_yes`.

Reply:
Yes good point. It looks like there's no reason for prophet to return a None prediction. Microchain has retry logic for errors too, so if we raise then the agent can retry the prediction.